### DISCUSSION TITLE
jsonable_encoder is giving error when converting results of sqlalchemy2 execute statement

### DISCUSSION THREAD

**Discussion by faizanazim11** [fastapi/fastapi#9330] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def fetch_query(self, query):
        try:
            result = self.session.execute(query)
            output = [jsonable_encoder(x) for x in result]
            self.session.close()
            return output
        except Exception as e:
            logging.error(f"Error occurred during execute_query: {e}")
```

### Description

When fetching results from execute statement of sqlalchemy2 and trying to convert it into dictionary objects using jsonable_encoder the code is giving error as below:

ValueError: [TypeError('cannot convert dictionary update sequence element #0 to a sequence'), TypeError('vars() argument must have __dict__ attribute')]

Note: Single object in results when doing a dict(obj) is giving the dictionary in sqlalchemy1.4, which is absent in sqlalchemy2. Also there is a method called _asdict in obj which returns the dict object.

For specifications of version of libraries:
Psycopg: 3
sqlalchemy: 2.0.0
fastapi: 0.95.0

This method is working with sqlalchemy 1.4

### Operating System

Windows

### FastAPI Version

0.95.0

### Python Version

3.9.13

**Comment by abhint** [comment]
Can you provide more code? 😃

**Comment by iudeen** [comment]
The reason seems to be because `sqlachemy.row.Row` object does not have `__dict__`. To get the dictionary from `Row`, obj._mapping` should be used. Not sure if we should add that in FastAPI's jsonable encoder or add `__dict__` in SQLAlchemy..

**Reply by mayurpande** [reply]
Any update on this, `sqlalchemy.row.Row` does have a `_asdict` method but not sure we should be using it.
